File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/AddSourceSinkTaskCommandMsg.js
const GraphBuildingCommandMsg = require('./GraphBuildingCommandMsg.js').GraphBuildingCommandMsg;

class AddSourceSinkTaskCommandMsg extends GraphBuildingCommandMsg
{
    constructor(taskName, taskTemporaryId, additionalInfo)
    {
        super();
        this.jsonMessage["taskName"] = taskName;
        this.jsonMessage["taskTemporaryId"] = taskTemporaryId;
        this.jsonMessage["additionalInfo"] = additionalInfo;
    }
}

module.exports = {AddSourceSinkTaskCommandMessage: AddSourceSinkTaskCommandMsg};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/AddSourceSinkTaskCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class AddSourceSinkTaskCommandMsg extends AddTaskCommandMsg
{
    private String additionalInfo;

    public AddSourceSinkTaskCommandMsg(String taskName, long taskTemporaryId, String additionalInfo)
    {
        super(taskName, taskTemporaryId);
        this.additionalInfo = additionalInfo;
    }

    public String getAdditionalInfo()
    {
        return additionalInfo;
    }
}
